### ISSUE TITLE
Double file extension making destroy action blow up

### ISSUE THREAD

**Issue by knuckolls** [ljfranklin/terraform-resource#31]
When testing the destroy action it works but returns as a failure because of an error message like this.

```
2017/03/30 20:10:58 State file does not exist with key 'tf-sandboxing.tfstate.tfstate'.
If you intended to run the `destroy` action, add `put.get_params.action: destroy`.
This is a temporary requirement until Concourse supports a `delete` step.
```

We think we have a fix and there may be a PR incoming. That double .tfstate looks like a bug.

**Comment by ljfranklin** [comment]
@knuckolls Double `tfstate` does look strange, but did you add the `put.get_params.action: destroy` field as described in the error message?

**Comment by knuckolls** [comment]
We did indeed. We have a fix but it is untested, I can open a PR to show you.

**Comment by knuckolls** [comment]
#32 is the PR that fixes this issue, theoretically

**Comment by ljfranklin** [comment]
@knuckolls Does your environment name (`put.params.env_name`) end with `.tfstate`?

**Comment by knuckolls** [comment]
It does not

**Comment by knuckolls** [comment]
Thank you for a quick response btw

**Comment by ljfranklin** [comment]
@knuckolls seems like a symptom of something else going wrong. Your PR shows that the filepath is constructed by adding `.tfstate` to the end of your env name, which seems to indicate that your env name does contain `.tfstate`. I env_name should show up in the UI somewhere. Going offline for a bit but will catch up later.

**Comment by knuckolls** [comment]
<img width="1613" alt="screen shot 2017-03-30 at 7 07 24 pm" src="https://cloud.githubusercontent.com/assets/91188/24531125/4732035e-157c-11e7-9d74-dffa326ad51a.png">

I believe this shows that the env_name does not in fact have .tfstate at the end.

**Comment by knuckolls** [comment]
Actually, that's in the apply stage, this is the failing one which does indeed have the .tfstate in the env_name, but that's not what's in the pipeline.yml.

<img width="1619" alt="screen shot 2017-03-30 at 7 09 54 pm" src="https://cloud.githubusercontent.com/assets/91188/24531179/9b8e3c74-157c-11e7-8334-15b65cfa0ede.png">


```
- name: job-destroy-terraform
  plan:
    - get: resource-pipelines
    - put: resource-terraform
      params:
        env_name: tf-sandboxing
        action: destroy
        get_params:
          action: destroy
          
```

**Comment by ljfranklin** [comment]
@knuckolls haven't forgotten about this, still need time to look into how it go into this state.

**Comment by knuckolls** [comment]
I appreciate you looking into it.

**Comment by ljfranklin** [comment]
@knuckolls think I've got it. For the failing destroy, you need to move `get_params` back one indent to make it even with `params`:

```
   - put: resource-terraform
      params:
        env_name: tf-sandboxing
        action: destroy
      get_params:
        action: destroy
```

I think that should get the error out of the way.

For the `tfstate` in the env name, I'm thinking it might be an bad interaction between deleting an environment that's already in a "tainted" state. If an apply fails, the resource uploads the statefile with a `.tainted` extension to make sure `get`s don't accidentally pull it. Did your previous destroy job fail and print a message about uploading a tainted version?

**Comment by ljfranklin** [comment]
@knuckolls I think this should be fixed as part of a331312c2b2d4622b7288cc404e176c5b0a65d87. Seemed to pop up if you tried to delete an environment after a failed update. Please re-open if you see it again.
